airbyte-cdk/java/airbyte-cdk/db-sources/src/testFixtures/java/io/airbyte/cdk/testutils/PostgresTestDatabase.java

```diff
@@ -40,18 +40,51 @@ public class PostgresTestDatabase implements AutoCloseable {
 
   static private final Logger LOGGER = LoggerFactory.getLogger(PostgresTestDatabase.class);
 
+  public enum PostgresImage {
+
+    POSTGRES_12_BULLSEYE("postgres:12-bullseye"),
+    POSTGRES_16_BULLSEYE("postgres:16-bullseye"),
+    POSTGRES_16_ALPINE("postgres:16-alpine"),
+    POSTGRES_9_ALPINE("postgres:9-alpine"),
+    ;
+
+    private final String imageName;
+
+    PostgresImage(String imageName) {
+      this.imageName = imageName;
+    }
+
+  }
+
+  public enum PostgresImageLayer {
+
+    ASCII(ContainerFactory::withASCII),
+    CONF(ContainerFactory::withConf),
+    NETWORK(ContainerFactory::withNetwork),
+    CERT(ContainerFactory::withCert),
+    WAL_LEVEL_LOGICAL(ContainerFactory::withWalLevelLogical),
+    ;
+
+    final Consumer<ContainerFactory> method;
+
+    PostgresImageLayer(Consumer<ContainerFactory> method) {
+      this.method = method;
+    }
+
+  }
+
   /**
    * Create a new {@link PostgresTestDatabase} instance.
    *
-   * @param imageName base image to use for the underlying {@link PostgreSQLContainer}.
+   * @param postgresImage base image to use for the underlying {@link PostgreSQLContainer}.
    * @param methods {@link ContainerFactory} methods that need to be called.
    * @return a new {@link PostgresTestDatabase} instance which may reuse a shared
    *         {@link PostgreSQLContainer}.
    */
-  static public PostgresTestDatabase make(String imageName, String... methods) {
+  static public PostgresTestDatabase make(PostgresImage postgresImage, PostgresImageLayer... methods) {
```

Comment:
If you take up my suggestion of s/PostgresImageLayer/PostgresContainerModifier/, please also rename s/methods/modifiers/